src/script/standard.cpp

```diff
@@ -68,6 +65,21 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, vector<vector<unsi
         return true;
     }
 
+    // Provably prunable, data-carrying output
+    //
+    // So long as script passes the IsUnspendable() test and all but the first
+    // byte passes the IsPushOnly() test we don't care what exactly is in the
+    // script.
+    if (scriptPubKey.size() >= 1 && scriptPubKey[0] == OP_RETURN
+                                 && scriptPubKey.size() <= nMaxDatacarrierBytes+2) // to account for the pushdata opcodes
```

Comment:
Also, nMaxDatacarrierBytes is only changed from its default value on init, again, why nMaxDatacarrierBytes+2 here instead of just nMaxDatacarrierBytes?